session.go

```diff
@@ -394,39 +415,109 @@ func (s *Session) sendNoWait(hdr header) error {
 	}
 }
 
-// send is a long running goroutine that sends data
-func (s *Session) send() {
+func pooledTimer(d time.Duration) (*time.Timer, func()) {
+	t := timerPool.Get()
+	timer := t.(*time.Timer)
+	timer.Reset(d)
+	var once sync.Once
```

Comment:
Didn't find a way to make `cancelFn` idempotent in a thread-safe manner without `sync.Once`, not because of the timer, but because of the pool management.